From 96eec9e35186d0d2c13ff4a20be048899bfe8cb8 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@windriver.com>
Date: Mon, 2 Feb 2009 12:01:02 -0500
Subject: [PATCH 2/4] ocd: powerpc hooks

Enable debug by turning on MSR debug enable (DE) bit. Not having
the DE bit set will result in hardware breakpoints and other
advanced debug register features to not work correctly.

Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
Signed-off-by: Mark Asselstine <mark.asselstine@windriver.com>
Signed-off-by: Yang Shi <yang.shi@windriver.com>
---
 arch/powerpc/include/asm/reg_booke.h | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/powerpc/include/asm/reg_booke.h b/arch/powerpc/include/asm/reg_booke.h
index b417de3..42cb1cb 100644
--- a/arch/powerpc/include/asm/reg_booke.h
+++ b/arch/powerpc/include/asm/reg_booke.h
@@ -37,8 +37,12 @@
 #define MSR_KERNEL	(MSR_ME|MSR_RI|MSR_IR|MSR_DR|MSR_CE)
 #define MSR_USER	(MSR_KERNEL|MSR_PR|MSR_EE)
 #else
-#define MSR_KERNEL	(MSR_ME|MSR_RI|MSR_CE)
-#define MSR_USER	(MSR_KERNEL|MSR_PR|MSR_EE)
+#ifdef CONFIG_WR_OCD_DEBUG
+#define MSR_KERNEL	(MSR_ME|MSR_RI|MSR_CE|MSR_DE)
+#else
+#define MSR_KERNEL     (MSR_ME|MSR_RI|MSR_CE)
+#endif	/* OCD */
+#define MSR_USER       (MSR_KERNEL|MSR_PR|MSR_EE)
 #endif
 
 /* Special Purpose Registers (SPRNs)*/
-- 
1.8.4.93.g57e4c17

